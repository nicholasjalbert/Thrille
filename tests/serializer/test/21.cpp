//test if debug info is working

#include <pthread.h>
#include <string>
#include <iostream>
#include <sstream>
#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>
#include <semaphore.h>
#include <execinfo.h>

using namespace std;

int x;
bool y;
sem_t sem;
pthread_t child, child2;
pthread_mutex_t mutex, mutex2;
pthread_cond_t cond, cond2;

void setup() {
    pthread_mutex_init(&mutex, NULL);
    pthread_mutex_init(&mutex2, NULL);
    pthread_cond_init(&cond, NULL);
    pthread_cond_init(&cond2, NULL);
}

void teardown() {
    pthread_mutex_destroy(&mutex);
    pthread_mutex_destroy(&mutex2);
    pthread_cond_destroy(&cond);
    pthread_cond_destroy(&cond2);
}

void * routine(void * arg) {
}

void bt() {

    void * ret = __builtin_return_address(0);
    ostringstream * str = new ostringstream();
    (*str) << "addr2line -e 21 " << ret;
    printf("%s\n", str->str().c_str());
    system(str->str().c_str());
}

int main(int argc, char *argv[]) {
    setup();
    bt();
    teardown();
}

