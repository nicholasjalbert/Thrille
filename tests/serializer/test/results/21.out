/home/jalbert/thrille/trunk/thrille/tests/serializer/test//21.cpp:51
addr2line -e 21 0x4022df
